src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -785,14 +785,15 @@ BoundNode bindSpreadElement(BoundCollectionExpressionSpreadElement element, Type
             return element;
         }
 
-        internal bool TryGetCollectionIterationType(ExpressionSyntax syntax, TypeSymbol collectionType, out TypeWithAnnotations iterationType)
+        internal bool TryGetCollectionExpressionIterationType(ExpressionSyntax syntax, TypeSymbol collectionType, out TypeWithAnnotations iterationType)
```

Comment:
>TryGetCollectionExpressionIterationType [](http://example.com/codeflow?start=22&length=39)

Consider leaving the name as is. This isn't really about an expression, this is about a type. For example, this method is used by "Params Collections" feature. There isn't even a collection expression in the picture for those scenarios. Another thing to consider, is the fact that I changed the signature of this method in [https://github.com/dotnet/roslyn/tree/features/ParamsCollections](https://github.com/dotnet/roslyn/tree/features/ParamsCollections) to use SyntaxNode instead of ExpressionSyntax due to the reason outlined above. The rename is likely to cause conflicts during integrations, which I would prefer to avoid.